### PR TITLE
Add support for enabling dynamic alsa loading in CMakeLists.txt

### PR DESCRIPTION
**Pull request by jcelerier**
It had been added to the source code a long time ago, but was not integrated to the
build system.

### CODE REVIEW

**Comment by jcelerier** [comment]
@RossBencina @philburk @dmitrykos if anyone could review that :) thanks !

File: CMakeLists.txt
Comment (dmitrykos) [inline CMakeLists.txt:310]: Why did you add these 2 lines? I believe they are not needed for the proposed functionality, only `PA_ALSA_DYNAMIC` needs to be defined without any other dependencies added. It has to be corrected.

**Review by dmitrykos — CHANGES_REQUESTED** [review]

File: CMakeLists.txt
Comment (jcelerier) [reply CMakeLists.txt:310]: no, on my machine it fails to link in that case as it does not find dlopen / dlsym.

**Review by dmitrykos — APPROVED** [review]

File: CMakeLists.txt
Comment (dmitrykos) [reply CMakeLists.txt:310]: Ok, understood, I overlooked that dl library is not used by PA by default. Then agree that these 2 lines are needed.

**Comment by dmitrykos** [comment]
Proposed addition looks fine to me, dynamic ALSA loading needs to be exposed for configuration via CMake.

**Review by RossBencina — APPROVED** [review]
Looks fine to me.

**Comment by jcelerier** [comment]
Hello, if anyone could merge that that would be very nice :)

**Comment by dmitrykos** [comment]
@jcelerier I was awaiting for an approval from @philburk . @philburk could we merge this pull request?

**Comment by RossBencina** [comment]
@dechamps could you give your opinion on this please?

**Comment by RossBencina** [comment]
@jcelerier please show some patience everyone here is volunteering in their spare time.

@dmitrykos thank you for waiting.

**Comment by jcelerier** [comment]
sure, no worries - I'm apparently too used to git workflows where two approvals enable merging, my bad.

**Comment by dechamps** [comment]
> @dechamps could you give your opinion on this please?

I do not work with PortAudio on Linux, only Windows, so I can't comment as to whether this makes sense or not.

**Comment by RossBencina** [comment]
@dechamps thanks for clarifying. no problems. I only ask because you are the only CMake-knowledgeable person around. Phil and I don't even know if it makes sense from a general CMake perspective.

**Comment by dechamps** [comment]
@RossBencina purely from a CMake perspective this seems fine to me.

**Comment by dmitrykos** [comment]
@RossBencina I have vast experience configuring build system with help of CMake and proposed change is perfectly fine. The only quirk was regarding linking libdl but PA is not linking it by default, so that question is also sorted out. It can be merged if you confirm.

**Review by philburk — APPROVED** [review]

**Comment by RossBencina** [comment]
@dechamps Thanks for your input, much appreciated.

@dmitrykos Your vast experience is appreciated but we're aiming to have *two* experienced people review each commit to master. Phil and I have almost no experience with CMake. I'm just trying to coordinate to get appropriate reviewers.

**Pull request merged**
